File: src/test/java/im/tox/tox4j/core/exceptions/ToxFileSendExceptionTest.java
package im.tox.tox4j.core.exceptions;

import im.tox.tox4j.ToxCoreImplTestBase;
import im.tox.tox4j.core.ToxCore;
import im.tox.tox4j.core.enums.ToxFileKind;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class ToxFileSendExceptionTest extends ToxCoreImplTestBase {

  @Test
  public void testFileSendNotConnected() throws Exception {
    try (ToxCore tox = newTox()) {
      int friendNumber = addFriends(tox, 1);
      try {
        tox.fileSend(friendNumber, ToxFileKind.DATA, 123, null, "filename".getBytes());
        fail();
      } catch (ToxFileSendException e) {
        assertEquals(ToxFileSendException.Code.FRIEND_NOT_CONNECTED, e.getCode());
      }
    }
  }

}

File: projects/tox4j/src/test/java/im/tox/tox4j/core/exceptions/ToxFileSendExceptionTest.scala
package im.tox.tox4j.core.exceptions

import im.tox.tox4j.core.ToxCoreConstants
import im.tox.tox4j.core.enums.ToxFileKind
import im.tox.tox4j.testing.ToxTestMixin
import org.scalatest.FunSuite

final class ToxFileSendExceptionTest extends FunSuite with ToxTestMixin {

  test("FileSendNotConnected") {
    interceptWithTox(ToxFileSendException.Code.FRIEND_NOT_CONNECTED)(
      _.fileSend(0, ToxFileKind.DATA, 123, null, "filename".getBytes)
    )
  }

  test("FileSendNotFound") {
    interceptWithTox(ToxFileSendException.Code.FRIEND_NOT_FOUND)(
      _.fileSend(1, ToxFileKind.DATA, 123, null, "filename".getBytes)
    )
  }

  test("FileSendNameTooLong") {
    interceptWithTox(ToxFileSendException.Code.NAME_TOO_LONG)(
      _.fileSend(0, ToxFileKind.DATA, 123, null, Array.ofDim[Byte](ToxCoreConstants.MaxFilenameLength + 1))
    )
  }

}
